core/src/main/java/org/arquillian/smart/testing/api/TestVerifier.java

```diff
@@ -0,0 +1,40 @@
+package org.arquillian.smart.testing.api;
+
+import org.arquillian.smart.testing.ClassNameExtractor;
+
+import java.nio.file.Path;
+
+/**
+ * Determines if a resource is an actual test for a given runtime (e.g. Surefire)
+ */
+public interface TestVerifier {
+
+    /**
+     * Checks whether the given path to a resource represents a test class or not
+     *
+     * @param resource A path to a resource to be checked
+     * @return Whether the given path to a resource represents a test class or not
+     */
+    default boolean isTest(Path resource) {
+        final String className = new ClassNameExtractor().extractFullyQualifiedName(resource);
+        return isTest(className);
+    }
+
+    /**
+     * Checks whether the given path to a resource represents a non-test class or not
+     *
+     * @param resource A path to a resource to be checked
+     * @return Whether the given path to a resource represents a non-test class or not
+     */
+    default boolean isCore(Path resource) {
+        return !isTest(resource);
```

Comment:
I'm not convinced if we need that really, because what does it really mean `Core`? But it's not to do in this PR, but to discuss.

Reply:
Yeah, I was asking myself the same question. I just took and kept the code there from the previous state...

Reply:
Sounds to me like refactoring task